test/react16_suggestors/react_style_maps_updater_test.dart

```diff
@@ -0,0 +1,569 @@
+                'margin': 25,
+              };
+            }
+          ''',
+        );
+      });
+
+      test('a null check with two variables', () {
+        testSuggestor(
+          expectedPatchCount: 3,
+          input: '''
+            $classSetupBoilerPlate
+            main() {
+              Foo()
+              ..style = {
+                'width': isWide ?? isNotWide,
+                'height': '40%',
+                'fontSize': '12',
+                'margin': '25',
+              };
+            }
+          ''',
+          expectedOutput: '''
+            $classSetupBoilerPlate
+            main() {
+              Foo()
+              ${manualVariableCheckComment(keysOfModdedValues: [
+            'width',
+            'fontSize',
+            'margin'
+          ])}
+              ..style = {
+                'width': isWide ?? isNotWide,
+                'height': '40%',
+                'fontSize': 12,
+                'margin': 25,
+              };
+            }
+          ''',
+        );
+      });
+
+      test('is an expression that has already been updated', () {
+        testSuggestor(
+          expectedPatchCount: 0,
+          input: '''
+            $classSetupBoilerPlate
+            main() {
+              Foo()
+              ${getCheckboxComment(keysOfModdedValues: [
+            'width',
+            'fontSize',
+            'margin'
+          ])}
+              ..style = {
+                'width': isWide ? 40 : 20,
+                'height': '40%',
+                'fontSize': 12,
+                'margin': 25,
+              };
+            }
+          ''',
+          expectedOutput: '''
+            $classSetupBoilerPlate
+            main() {
+              Foo()
+              ${getCheckboxComment(keysOfModdedValues: [
+            'width',
+            'fontSize',
+            'margin'
+          ])}
+              ..style = {
+                'width': isWide ? 40 : 20,
+                'height': '40%',
+                'fontSize': 12,
+                'margin': 25,
+              };
+            }
+          ''',
+        );
+      });
+
+      test('a function call', () {
+        testSuggestor(
+          expectedPatchCount: 1,
+          input: '''
+            $classSetupBoilerPlate
+            main() {
+              Map getStyleMap() {
+                return {
+                  'width': '40px',
+                };
+              }
+            
+              Foo()
+              ..style = getStyleMap();
+            }
+          ''',
+          expectedOutput: '''
+            $classSetupBoilerPlate
+            main() {
+              Map getStyleMap() {
+                return {
+                  'width': '40px',
+                };
+              }
+            
+              Foo()
+              ${getFunctionComment()}
+              ..style = getStyleMap();
+            }
+          ''',
+        );
+      });
+
+      test('is a nested component', () {
```

Comment:
could we call this 'is a component as prop value`?

then add one that 'is a nested component' or 'is variadic children'
that has something like this: 
```
      test('is variadic children', () {
        testSuggestor(
          expectedPatchCount: 4,
          input: '''
            main() {
              (Foo()
              ..id = 'number1'
              ..style = {'width': '40',}
              )(
                (Bar()
                ..style = {'width': '60',}
                )(),
              );
            }
          ''',
          expectedOutput: '''
            main() {
              (Foo()
              ..id = 'number1'
              ${getCheckboxComment(keysOfModdedValues: ['width'])}
              ..style = {'width': 40,})(
                (Bar()
                  ${getCheckboxComment(keysOfModdedValues: ['width'])}
                  ..style = {
                    'width': 60,
                  })(),
              );
            }
          ''',
        );
      });
```

Reply:
That's a good idea, thank you for writing the full test!